c/misra/src/rules/RULE-5-4/MacroIdentifiersNotDistinct.ql

```diff
@@ -0,0 +1,33 @@
+/**
+ * @id c/misra/macro-identifiers-not-distinct
+ * @name RULE-5-4: Macro identifiers shall be distinct
+ * @description Declaring multiple macros with the same name leads to undefined behaviour.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-5-4
+ *       correctness
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+from Macro m, Macro m2
+where
+  not isExcluded(m, Declarations1Package::macroIdentifiersNotDistinctQuery()) and
+  not m = m2 and
+  (
+    //C99 states the first 63 characters of macro identifiers are significant
+    //C90 states the first 31 characters of macro identifiers are significant and is not currently considered by this rule
+    //ie an identifier differing on the 32nd character would be indistinct for C90 but distinct for C99
+    //and is currently not reported by this rule
+    if m.getName().length() >= 64
+    then m.getName().substring(0, 62) = m2.getName().substring(0, 62)
```

Comment:
1. I think you want 63 not 62, as the end is an exclusive offset.
 2. You can use `prefix(63)` instead of `substring`.

Reply:
another good catch, done in both this and 5-1!